src/screens/planActivity/PlanForm.tsx

```diff
@@ -3,36 +3,28 @@ import React from 'react';
 import { NavigationInjectedProps, withNavigation } from 'react-navigation';
 
 import { Plan } from 'models';
-import { PlanFormControls } from './PlanFormControls';
+import PlanFormControls from './PlanFormControls';
 
 export interface PlanFormData {
   planInput: string;
 }
 
-interface State {
-  plan: Plan;
-}
-
 const FORM_INITIAL_VALUES: PlanFormData = {
   planInput: '',
 };
 
-export class PlanForm extends React.PureComponent<NavigationInjectedProps, State> {
-  state: State = {
-    plan: new Plan(),
-  };
-
-  async setPlanState(planInput: string) {
+export class PlanForm extends React.PureComponent<NavigationInjectedProps> {
+  async createPlan(planInput: string) {
```

Comment:
You can move that function higher to the `PlanActivityScreen`, change the name of `PlanFormControls` to `PlanForm`, and handle the `onSubmit` in higher component. Going that way, you will resign from using `withNavigation` in `PlanFormControls` and `PlanActivityScreen`, and pass the data only through react props. It will improve code readibility and simplify the flow.